### PR TITLE
Remove global symbols: Avoid using the global namespace if possible

### LINKED ISSUE #15612: Reduce the number of global symbols used
The `bitcoind` binary contains more than 79 global symbols as reported by `nm` (in the BSS data section and initialized data section, somewhat filtered - see oneliner below):

```
$ for SYMBOL in $(nm src/bitcoind | grep -E ' [BD] ' | c++filt | cut -f3- -d' ' | grep -v @ | grep -v : | sort | grep '[a-z]' | sort -u | grep -vE '^_'); do
     REFERENCES=$(git grep -lE "([^a-zA-Z]|^)${SYMBOL}([^a-zA-Z]|\$)" -- "*.cpp" "*.h")
     N_REFERENCES=$(wc -l <<< "${REFERENCES}")
     echo -n "Found ${N_REFERENCES} use(s) of ${SYMBOL}: "
     sort -u <<< "${REFERENCES}" | tr "\n" " "
     echo
  done
Found 1 use(s) of boundSockets: src/httpserver.cpp
Found 25 use(s) of chainActive: src/bench/block_assemble.cpp src/bench/duplicate_inputs.cpp src/index/base.cpp src/index/txindex.cpp src/init.cpp src/interfaces/chain.cpp src/interfaces/node.cpp src/miner.cpp src/net_processing.cpp src/policy/fees.cpp src/qt/test/wallettests.cpp src/rest.cpp src/rpc/blockchain.cpp src/rpc/mining.cpp src/rpc/rawtransaction.cpp src/test/denialofservice_tests.cpp src/test/miner_tests.cpp src/test/test_bitcoin.cpp src/test/txvalidationcache_tests.cpp src/test/validation_block_tests.cpp src/txmempool.h src/validation.cpp src/validation.h src/wallet/test/wallet_tests.cpp src/wallet/wallet.cpp
Found 43 use(s) of cs_main: src/bench/block_assemble.cpp src/bench/duplicate_inputs.cpp src/bench/mempool_eviction.cpp src/bench/rpc_mempool.cpp src/index/base.cpp src/index/txindex.cpp src/init.cpp src/interfaces/chain.cpp src/interfaces/node.cpp src/interfaces/wallet.cpp src/miner.cpp src/net.h src/net_processing.cpp src/net_processing.h src/node/transaction.cpp src/qt/clientmodel.cpp src/qt/peertablemodel.cpp src/rest.cpp src/rpc/blockchain.cpp src/rpc/mining.cpp src/rpc/misc.cpp src/rpc/net.cpp src/rpc/rawtransaction.cpp src/test/blockencodings_tests.cpp src/test/denialofservice_tests.cpp src/test/mempool_tests.cpp src/test/miner_tests.cpp src/test/policyestimator_tests.cpp src/test/script_p2sh_tests.cpp src/test/test_bitcoin.cpp src/test/transaction_tests.cpp src/test/txvalidationcache_tests.cpp src/test/txvalidation_tests.cpp src/test/validation_block_tests.cpp src/txmempool.h src/validation.cpp src/validation.h src/validationinterface.cpp src/validationinterface.h src/wallet/test/wallet_tests.cpp src/wallet/wallet.cpp src/wallet/wallet.h src/zmq/zmqpublishnotifier.cpp
Found 4 use(s) of cs_mapLocalHost: src/net.cpp src/net.h src/rpc/net.cpp src/test/net_tests.cpp
Found 1 use(s) of cs_warnings: src/warnings.cpp
Found 7 use(s) of dustRelayFee: src/init.cpp src/interfaces/node.cpp src/policy/policy.cpp src/policy/policy.h src/test/transaction_tests.cpp src/wallet/fees.cpp src/wallet/wallet.cpp
Found 1 use(s) of eventHTTP: src/httpserver.cpp
Found 4 use(s) of fAcceptDatacarrier: src/init.cpp src/policy/policy.cpp src/script/standard.cpp src/script/standard.h
Found 4 use(s) of fCheckBlockIndex: src/init.cpp src/test/test_bitcoin.cpp src/validation.cpp src/validation.h
Found 4 use(s) of fCheckpointsEnabled: src/init.cpp src/test/miner_tests.cpp src/validation.cpp src/validation.h
Found 3 use(s) of fDiscover: src/init.cpp src/net.cpp src/net.h
Found 6 use(s) of feeEstimator: src/init.cpp src/interfaces/chain.cpp src/interfaces/node.cpp src/rpc/mining.cpp src/validation.cpp src/validation.h
Found 3 use(s) of fEnableReplacement: src/init.cpp src/validation.cpp src/validation.h
Found 1 use(s) of fFeeEstimatesInitialized: src/init.cpp
Found 3 use(s) of fHavePruned: src/init.cpp src/validation.cpp src/validation.h
Found 5 use(s) of fImporting: src/init.cpp src/interfaces/node.cpp src/net_processing.cpp src/validation.cpp src/validation.h
Found 4 use(s) of fIsBareMultisigStd: src/init.cpp src/policy/policy.cpp src/validation.cpp src/validation.h
Found 1 use(s) of fLargeWorkForkFound: src/warnings.cpp
Found 1 use(s) of fLargeWorkInvalidChainFound: src/warnings.cpp
Found 5 use(s) of fListen: src/init.cpp src/net.cpp src/net.h src/net_processing.cpp src/qt/optionsmodel.cpp
Found 6 use(s) of fLogIPs: src/init.cpp src/logging.cpp src/logging.h src/net.cpp src/net_processing.cpp src/rpc/server.cpp
Found 4 use(s) of fNameLookup: src/init.cpp src/netbase.cpp src/netbase.h src/net.cpp
Found 6 use(s) of fPruneMode: src/init.cpp src/interfaces/chain.cpp src/net_processing.cpp src/rpc/blockchain.cpp src/validation.cpp src/validation.h
Found 5 use(s) of fReindex: src/init.cpp src/interfaces/node.cpp src/net_processing.cpp src/validation.cpp src/validation.h
Found 5 use(s) of fRelayTxes: src/init.cpp src/net.cpp src/net.h src/net_processing.cpp src/rpc/net.cpp
Found 5 use(s) of fRequireStandard: src/chainparams.cpp src/chainparams.h src/init.cpp src/validation.cpp src/validation.h
Found 51 use(s) of gArgs: src/bench/bench_bitcoin.cpp src/bitcoin-cli.cpp src/bitcoind.cpp src/bitcoin-tx.cpp src/bitcoin-wallet.cpp src/chainparamsbase.cpp src/chainparams.cpp src/dbwrapper.cpp src/dummywallet.cpp src/httprpc.cpp src/httpserver.cpp src/init.cpp src/init.h src/interfaces/chain.cpp src/interfaces/node.cpp src/miner.cpp src/net.cpp src/net_processing.cpp src/qt/bitcoin.cpp src/qt/guiutil.cpp src/qt/intro.cpp src/qt/optionsmodel.cpp src/qt/paymentrequestplus.cpp src/qt/paymentserver.cpp src/qt/test/test_main.cpp src/qt/utilitydialog.cpp src/qt/walletmodel.cpp src/rpc/blockchain.cpp src/rpc/protocol.cpp src/rpc/server.cpp src/script/sigcache.cpp src/test/denialofservice_tests.cpp src/test/getarg_tests.cpp src/test/net_tests.cpp src/test/test_bitcoin.cpp src/timedata.cpp src/torcontrol.cpp src/txdb.cpp src/util/system.cpp src/util/system.h src/validation.cpp src/wallet/coincontrol.cpp src/wallet/db.cpp src/wallet/feebumper.cpp src/wallet/init.cpp src/wallet/test/init_test_fixture.cpp src/wallet/test/init_tests.cpp src/wallet/wallet.cpp src/wallet/walletdb.cpp src/wallet/walletutil.cpp src/zmq/zmqnotificationinterface.cpp
Found 6 use(s) of g_banman: src/banman.h src/init.cpp src/interfaces/node.cpp src/net.h src/rpc/net.cpp src/test/test_bitcoin.cpp
Found 4 use(s) of g_best_block: src/init.cpp src/rpc/mining.cpp src/validation.cpp src/validation.h
Found 4 use(s) of g_best_block_cv: src/init.cpp src/rpc/mining.cpp src/validation.cpp src/validation.h
Found 3 use(s) of g_best_block_mutex: src/rpc/mining.cpp src/validation.cpp src/validation.h
Found 1 use(s) of g_chainstate: src/validation.cpp
Found 8 use(s) of g_connman: src/init.cpp src/interfaces/chain.cpp src/interfaces/node.cpp src/net.h src/node/transaction.cpp src/rpc/mining.cpp src/rpc/net.cpp src/test/test_bitcoin.cpp
Found 3 use(s) of g_cs_orphans: src/net.h src/net_processing.cpp src/test/denialofservice_tests.cpp
Found 4 use(s) of g_is_mempool_loaded: src/init.cpp src/rpc/blockchain.cpp src/validation.cpp src/validation.h
Found 4 use(s) of g_mock_deterministic_tests: src/random.cpp src/test/bloom_tests.cpp src/test/random_tests.cpp src/test/test_bitcoin.h
Found 6 use(s) of g_rpc_interfaces: src/init.cpp src/rpc/rawtransaction.cpp src/rpc/util.cpp src/rpc/util.h src/test/rpc_tests.cpp src/wallet/rpcwallet.cpp
Found 7 use(s) of g_txindex: src/index/txindex.cpp src/index/txindex.h src/init.cpp src/rest.cpp src/rpc/blockchain.cpp src/rpc/rawtransaction.cpp src/validation.cpp
Found 1 use(s) of g_ui_signals: src/ui_interface.cpp
Found 5 use(s) of g_wallet_init_interface: src/dummywallet.cpp src/httprpc.cpp src/init.cpp src/wallet/init.cpp src/walletinitinterface.h
Found 4 use(s) of g_zmq_notification_interface: src/init.cpp src/zmq/zmqnotificationinterface.cpp src/zmq/zmqnotificationinterface.h src/zmq/zmqrpc.cpp
Found 3 use(s) of hashAssumeValid: src/init.cpp src/validation.cpp src/validation.h
Found 8 use(s) of incrementalRelayFee: src/init.cpp src/policy/policy.cpp src/policy/policy.h src/rpc/net.cpp src/txmempool.cpp src/txmempool.h src/validation.cpp src/wallet/feebumper.cpp
Found 1 use(s) of instance_of_cmaincleanup: src/validation.cpp
Found 1 use(s) of instance_of_cnetcleanup: src/net.cpp
Found 1 use(s) of instance_of_cnetprocessingcleanup: src/net_processing.cpp
Found 8 use(s) of mapBlockIndex: src/init.cpp src/interfaces/wallet.cpp src/net_processing.cpp src/rpc/blockchain.cpp src/txdb.cpp src/validation.cpp src/validation.h src/wallet/test/wallet_tests.cpp
Found 4 use(s) of mapLocalHost: src/net.cpp src/net.h src/rpc/net.cpp src/test/net_tests.cpp
Found 2 use(s) of mapOrphanTransactions: src/net_processing.cpp src/test/denialofservice_tests.cpp
Found 11 use(s) of maxTxFee: src/init.cpp src/interfaces/chain.cpp src/interfaces/chain.h src/interfaces/node.cpp src/qt/walletmodel.cpp src/rpc/rawtransaction.cpp src/validation.cpp src/validation.h src/wallet/feebumper.cpp src/wallet/fees.cpp src/wallet/wallet.cpp
Found 55 use(s) of mempool: src/bench/block_assemble.cpp src/bench/mempool_eviction.cpp src/blockencodings.cpp src/blockencodings.h src/init.cpp src/interfaces/chain.cpp src/interfaces/chain.h src/interfaces/node.cpp src/interfaces/node.h src/interfaces/wallet.cpp src/interfaces/wallet.h src/logging.cpp src/miner.cpp src/miner.h src/net.h src/net_processing.cpp src/node/transaction.cpp src/policy/fees.cpp src/policy/fees.h src/policy/policy.h src/policy/rbf.cpp src/policy/rbf.h src/protocol.cpp src/protocol.h src/qt/bitcoinstrings.cpp src/qt/coincontroldialog.cpp src/qt/rpcconsole.h src/qt/test/rpcnestedtests.cpp src/qt/transactionrecord.h src/rest.cpp src/rpc/blockchain.cpp src/rpc/client.cpp src/rpc/mining.cpp src/rpc/misc.cpp src/rpc/net.cpp src/rpc/rawtransaction.cpp src/test/blockencodings_tests.cpp src/test/mempool_tests.cpp src/test/miner_tests.cpp src/test/policyestimator_tests.cpp src/test/test_bitcoin.cpp src/test/txvalidationcache_tests.cpp src/test/txvalidation_tests.cpp src/txmempool.cpp src/txmempool.h src/validation.cpp src/validation.h src/validationinterface.h src/wallet/feebumper.cpp src/wallet/feebumper.h src/wallet/fees.cpp src/wallet/init.cpp src/wallet/rpcwallet.cpp src/wallet/wallet.cpp src/wallet/wallet.h
Found 11 use(s) of minRelayTxFee: src/init.cpp src/net_processing.cpp src/policy/policy.h src/rpc/blockchain.cpp src/rpc/net.cpp src/validation.cpp src/validation.h src/wallet/fees.cpp src/wallet/init.cpp src/wallet/rpcwallet.cpp src/wallet/wallet.cpp
Found 3 use(s) of nBytesPerSigOp: src/init.cpp src/policy/policy.cpp src/policy/policy.h
Found 3 use(s) of nCoinCacheUsage: src/init.cpp src/validation.cpp src/validation.h
Found 4 use(s) of nConnectTimeout: src/init.cpp src/netbase.cpp src/netbase.h src/net.cpp
Found 4 use(s) of nMaxDatacarrierBytes: src/init.cpp src/policy/policy.cpp src/script/standard.cpp src/script/standard.h
Found 3 use(s) of nMaxTipAge: src/init.cpp src/validation.cpp src/validation.h
Found 6 use(s) of nMinimumChainWork: src/chainparams.cpp src/consensus/params.h src/init.cpp src/net_processing.cpp src/validation.cpp src/validation.h
Found 4 use(s) of nPruneTarget: src/init.cpp src/rpc/blockchain.cpp src/validation.cpp src/validation.h
Found 5 use(s) of nScriptCheckThreads: src/init.cpp src/test/checkqueue_tests.cpp src/test/test_bitcoin.cpp src/validation.cpp src/validation.h
Found 13 use(s) of NullUniValue: src/bitcoin-cli.cpp src/httprpc.cpp src/rpc/blockchain.cpp src/rpc/mining.cpp src/rpc/misc.cpp src/rpc/net.cpp src/rpc/protocol.cpp src/rpc/server.cpp src/rpc/server.h src/univalue/include/univalue.h src/univalue/lib/univalue.cpp src/wallet/rpcdump.cpp src/wallet/rpcwallet.cpp
Found 1 use(s) of pathHandlers: src/httpserver.cpp
Found 7 use(s) of pblocktree: src/bench/block_assemble.cpp src/bench/duplicate_inputs.cpp src/index/txindex.cpp src/init.cpp src/test/test_bitcoin.cpp src/validation.cpp src/validation.h
Found 7 use(s) of pcoinsdbview: src/bench/block_assemble.cpp src/bench/duplicate_inputs.cpp src/init.cpp src/rpc/blockchain.cpp src/test/test_bitcoin.cpp src/validation.cpp src/validation.h
Found 15 use(s) of pcoinsTip: src/bench/block_assemble.cpp src/bench/duplicate_inputs.cpp src/init.cpp src/interfaces/node.cpp src/net_processing.cpp src/node/transaction.cpp src/rest.cpp src/rpc/blockchain.cpp src/rpc/rawtransaction.cpp src/test/miner_tests.cpp src/test/test_bitcoin.cpp src/test/txvalidationcache_tests.cpp src/txmempool.h src/validation.cpp src/validation.h
Found 2 use(s) of peerLogic: src/init.cpp src/test/denialofservice_tests.cpp
Found 1 use(s) of pindexBestForkBase: src/validation.cpp
Found 1 use(s) of pindexBestForkTip: src/validation.cpp
Found 5 use(s) of pindexBestHeader: src/interfaces/node.cpp src/net_processing.cpp src/rpc/blockchain.cpp src/validation.cpp src/validation.h
Found 3 use(s) of secp256k1_nonce_function_default: src/secp256k1/include/secp256k1.h src/secp256k1/include/secp256k1_recovery.h src/secp256k1/src/modules/recovery/main_impl.h
Found 2 use(s) of secp256k1_nonce_function_rfc6979: src/key.cpp src/secp256k1/include/secp256k1.h
Found 11 use(s) of tableRPC: src/httprpc.cpp src/init.cpp src/interfaces/node.cpp src/interfaces/wallet.cpp src/qt/test/rpcnestedtests.cpp src/rpc/register.h src/rpc/server.cpp src/rpc/server.h src/test/rpc_tests.cpp src/test/test_bitcoin.cpp src/wallet/test/wallet_test_fixture.cpp
Found 1 use(s) of threadHTTP: src/httpserver.cpp
Found 17 use(s) of uiInterface: src/httprpc.cpp src/httpserver.cpp src/index/base.cpp src/index/txindex.cpp src/init.cpp src/interfaces/chain.cpp src/interfaces/node.cpp src/net.cpp src/net.h src/noui.cpp src/timedata.cpp src/txdb.cpp src/ui_interface.cpp src/ui_interface.h src/validation.cpp src/wallet/init.cpp src/wallet/rpcdump.cpp
Found 3 use(s) of versionbitscache: src/rpc/mining.cpp src/validation.cpp src/validation.h
Found 6 use(s) of VersionBitsDeploymentInfo: src/chainparams.cpp src/consensus/params.h src/rpc/blockchain.cpp src/rpc/mining.cpp src/versionbitsinfo.cpp src/versionbitsinfo.h
$ for SYMBOL in $(nm src/bitcoind | c++filt | grep -E ' B .*[a-z]\[abi:cxx11\]' | cut -f3- -d' ' | cut -f1 -d'[' | sort -u); do
       REFERENCES=$(git grep -lE "([^a-zA-Z]|^)${SYMBOL}([^a-zA-Z]|\$)" -- "*.cpp" "*.h")
       N_REFERENCES=$(wc -l <<< "${REFERENCES}")
       echo -n "Found ${N_REFERENCES} use(s) of ${SYMBOL}: "
       sort -u <<< "${REFERENCES}" | tr "\n" " "
       echo
  done
Found 5 use(s) of strMessageMagic: src/qt/signverifymessagedialog.cpp src/rpc/misc.cpp src/validation.cpp src/validation.h src/wallet/rpcwallet.cpp
Found 1 use(s) of strMiscWarning: src/warnings.cpp
Found 6 use(s) of strSubVersion: src/init.cpp src/net.cpp src/net.h src/net_processing.cpp src/qt/clientmodel.cpp src/rpc/net.cpp
```

Most of these symbols are global (external) for good reasons.

However, judging from the output above the following 16 symbols do not have to be global (external):

```
Found 1 use(s) of boundSockets: src/httpserver.cpp
Found 1 use(s) of cs_warnings: src/warnings.cpp
Found 1 use(s) of eventHTTP: src/httpserver.cpp
Found 1 use(s) of fFeeEstimatesInitialized: src/init.cpp
Found 1 use(s) of fLargeWorkForkFound: src/warnings.cpp
Found 1 use(s) of fLargeWorkInvalidChainFound: src/warnings.cpp
Found 1 use(s) of g_chainstate: src/validation.cpp
Found 1 use(s) of g_ui_signals: src/ui_interface.cpp
Found 1 use(s) of instance_of_cmaincleanup: src/validation.cpp
Found 1 use(s) of instance_of_cnetcleanup: src/net.cpp
Found 1 use(s) of instance_of_cnetprocessingcleanup: src/net_processing.cpp
Found 1 use(s) of pathHandlers: src/httpserver.cpp
Found 1 use(s) of pindexBestForkBase: src/validation.cpp
Found 1 use(s) of pindexBestForkTip: src/validation.cpp
Found 1 use(s) of strMiscWarning: src/warnings.cpp
Found 1 use(s) of threadHTTP: src/httpserver.cpp
```

### PR DESCRIPTION
**Pull request by practicalswift**
Remove global symbols: Avoid using the global namespace if possible.

Partially resolves #15612 ("Reduce the number of global symbols used").

Change in global symbols as reported by `nm bitcoind` before vs after:

```
$ diff -u <(nm src/bitcoind-before | c++filt | grep -E '^[0-9a-f]+ [A-Z] ' | cut -f3- -d' ' | sort -u) \
          <(nm src/bitcoind-after  | c++filt | grep -E '^[0-9a-f]+ [A-Z] ' | cut -f3- -d' ' | sort -u) \
          | grep -E '^[+-][^+-]'
-boundSockets
-cs_warnings
-eventHTTP
-fFeeEstimatesInitialized
-fLargeWorkForkFound
-fLargeWorkInvalidChainFound
-pathHandlers
-strMiscWarning[abi:cxx11]
-threadHTTP
```

### CODE REVIEW

File: src/httpserver.cpp [hunk 1]
```diff
@@ -137,16 +137,20 @@ struct HTTPPathHandler
...
-//! Handlers for (sub)paths
-std::vector<HTTPPathHandler> pathHandlers;
-//! Bound listening sockets
-std::vector<evhttp_bound_socket *> boundSockets;
+namespace {
+    //! Handlers for (sub)paths
+    std::vector<HTTPPathHandler> pathHandlers;
+    //! Bound listening sockets
+    std::vector<evhttp_bound_socket *> boundSockets;
+}
```
Comment (MarcoFalke) [inline src/httpserver.cpp]: Why can't you wrap everything above into a namespace? Also should not change the indentation. I'd suggest to use clang-format

Comment (practicalswift) [reply src/httpserver.cpp]: Good point! Fixed!

With this change `HTTPPathHandler`, `WorkQueue` and `HTTPWorkItem` are no longer global.

Please re-review!

**Comment by promag** [comment]
Concept ACK, change looks good.

**Comment by MarcoFalke** [comment]
utACK b112097f8e5f96e96db47e2eef9870d6f0431677. Also checked that the binary is 9kB smaller.

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15622/commits/b112097f8e5f96e96db47e2eef9870d6f0431677

File: src/init.cpp [hunk 2]
```diff
@@ -71,13 +71,15 @@
 #include <zmq/zmqrpc.h>
 #endif
 
+namespace {
 bool fFeeEstimatesInitialized = false;
 static const bool DEFAULT_PROXYRANDOMIZE = true;
```
Comment (promag) [inline src/init.cpp]: I think you could drop `static`?

Comment (practicalswift) [reply src/init.cpp]: Yes, absolutely but I didn't want to change any existing lines to make review super trivial. Let me know if that decision is blocking an utACK from you and I'll consider reconsidering :-)

Comment (Empact) [reply src/init.cpp]: With or without the declaration seems they'll have internal linkage (via unnamed namespace) and static storage duration (via namespace scope).

Mildly in favor of removal on the grounds that a distinction without a difference in this case can be misleading.

> `static` When used in a declaration of an object, it specifies static storage duration. When used in a declaration at namespace scope, it specifies internal linkage.
> static storage duration. All objects declared at namespace scope (including global namespace) have this storage duration, plus those declared with static or extern.
https://en.cppreference.com/w/cpp/language/storage_duration

> Unnamed namespace definition. Its members have potential scope from their point of declaration to the end of the translation unit, and have internal linkage.
https://en.cppreference.com/w/cpp/language/namespace#Unnamed_namespaces

**Comment by practicalswift** [comment]
Now removing redundant `static`:s in unnamed namespaces as requested by @promag and @Empact.

Please re-review :-)

**Comment by promag** [comment]
utACK 9638da7.

**Comment by lucayepa** [comment]
I think we can add some language on this policy in doc/developer-notes.md. It will help new developers. I agree on a zero-globals policy, enforced with singletons, or with a "state" class that is a container of all the globals, or with a dedicated namespace. This way, every time a new global has to be added (rarely), we can see a clear change in a single place. In my mind this policy should not be enforced from now, but should be accepted, then written in doc/developer-notes.md or elsewhere, and enforced after a couple of years, when all the open PRs are already compliant.

I've just described this method of long grace periods in my [comment on #15465](https://github.com/bitcoin/bitcoin/issues/15465#issuecomment-475914641).

**Comment by practicalswift** [comment]
@lucayepa I would like to restrict the scope of this PR to these specific code changes which have already been (ut)ACK:ed.

**Comment by Empact** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/15622/commits/9638da76386ba1e6f610f26a5bc9b88825087bdc

File: src/init.cpp [hunk 3]
```diff
@@ -71,13 +71,15 @@
...
-static const bool DEFAULT_REST_ENABLE = false;
-static const bool DEFAULT_STOPAFTERBLOCKIMPORT = false;
+const bool DEFAULT_PROXYRANDOMIZE = true;
+const bool DEFAULT_REST_ENABLE = false;
+const bool DEFAULT_STOPAFTERBLOCKIMPORT = false;
 
 // Dump addresses to banlist.dat every 15 minutes (900s)
-static constexpr int DUMP_BANS_INTERVAL = 60 * 15;
+constexpr int DUMP_BANS_INTERVAL = 60 * 15;
+} // namespace
```
Comment (MarcoFalke) [inline src/init.cpp]: Why is this changed? What is the difference between `static const(expr) int` and `const int` in a namespace?

We don't change code for stylistic reasons.

Comment (MarcoFalke) [reply src/init.cpp]: Also, I cant see a reason to not have literals in the global namespace

Comment (sipa) [reply src/init.cpp]: static inside a anonymous namespace is redundant.

**Comment by sipa** [comment]
It's a bit confusing to call this"Remove globals"; a global inside a namespace is still a global.

**Comment by practicalswift** [comment]
@sipa I've now clarified that I this refers to global **symbols** :-)

File: src/init.cpp (hunk 3 above)
Comment (practicalswift) [reply src/init.cpp]: Please see also https://github.com/bitcoin/bitcoin/pull/15622#discussion_r267802687

Let me know if you find any reason to revert to the original version.

Comment (practicalswift) [reply src/init.cpp]: > Also, I cant see a reason to not have literals in the global namespace

If we want to share say `DUMP_BANS_INTERVAL` to another compilation unit the recommended way to do it is via `#include "init.h"`, no?

Given the amount of hate we get due to the said environmental impact of SHA256² we don't want to make things worse by polluting also the global namespace! :-)

Comment (MarcoFalke) [reply src/init.cpp]: >  static inside a anonymous namespace is redundant.

ftfy: anonymous namespace is redundant when the constexpr is already static

**Comment by practicalswift** [comment]
utACK:s from @Empact and @promag, one stale utACK from @MarcoFalke and zero NACK:s

@MarcoFalke, are we ready to move forward with this one to get a slightly less polluted global namespace? :-)

**Comment by ajtowns** [comment]
I would have thought it would be better to write these as `static ...` than `namespace { \n ... \n }` -- you can grep for the former relatively easily, but not really the latter, and it seems more consistent with the current style... It gives a +9-9 diff which seems better than +17-9...

Either way, I don't see the point of dropping `static` just because it's redundant; that seems like needless churn. After stripping the binary, i don't see any size difference in bitcoind amongst these variants on Linux with clang fwiw.

**Comment by practicalswift** [comment]
@ajtowns I've now reverted to the original version of the PR which is in accordance with your suggestions. Please re-review :-)

@promag @MarcoFalke @Empact Would you mind re-reviewing? :-)

File: src/warnings.cpp [hunk 4]
```diff
@@ -8,10 +8,10 @@
 #include <util/system.h>
 #include <warnings.h>
 
-CCriticalSection cs_warnings;
-std::string strMiscWarning GUARDED_BY(cs_warnings);
-bool fLargeWorkForkFound GUARDED_BY(cs_warnings) = false;
-bool fLargeWorkInvalidChainFound GUARDED_BY(cs_warnings) = false;
+static CCriticalSection cs_warnings;
```
Comment (MarcoFalke) [inline src/warnings.cpp]: style nit:

```suggestion
static RecursiveMutex cs_warnings;
```

Comment (practicalswift) [reply src/warnings.cpp]: Done!

**Comment by MarcoFalke** [comment]
Please squash the second commit, no need to have a separate commit for a single line stylistic fixup

**Comment by practicalswift** [comment]
@MarcoFalke Done! Please re-review :-)
